pythran/pythonic/numpy/partial_sum.hpp

```diff
@@ -13,24 +12,53 @@ namespace pythonic
   namespace numpy
   {
 
-    template <class Op>
-    template <class E, class F, class A>
-    A _partial_sum<Op, 1>::operator()(E e, F &o, A acc)
+    template <class Op, class A>
+    template <class E, class F>
+    A _partial_sum<Op, 1, A>::operator()(E e, F &o, A acc)
     {
       for (auto const &value : e) {
-        Op{}(acc, value);
+        acc = Op{}(acc, (A)value);
         *o = acc;
         ++o;
       }
       return acc;
     }
 
-    template <class Op, size_t N>
-    template <class E, class F, class A>
-    A _partial_sum<Op, N>::operator()(E e, F &o, A acc)
+    template <class Op, class A>
+    template <class E, class F>
+    A _partial_sum<Op, 1, A>::operator()(E e, F &o)
+    {
+      auto it_begin = e.begin();
+      A acc = *it_begin;
+      *o = acc;
+      ++o;
+      ++it_begin;
+      for (; it_begin < e.end(); ++it_begin) {
+        acc = Op{}(acc, (A)*it_begin);
+        *o = acc;
+        ++o;
```

Comment:
I'd rather put the `++o` inside the loop header `++it_begin, ++o` make things look more explicit to me.